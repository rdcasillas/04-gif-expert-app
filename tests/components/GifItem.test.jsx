import { GifItem } from "../../src/components/GifItem";
const { render, screen } = require("@testing-library/react");


describe('Pruebas en <GifItem/>', () => { 

    const title = 'Dragon Ball';
    const url= 'https://gifexpertapp.com/dragon-ball.gif';

    test('debe de hacer match con el snapshot', () => { 

        const { container } =  render( <GifItem title={ title } url={ url } />);

        expect( container ).toMatchSnapshot();
     });

     test('debe de mostrar la imagen con el URL y el ALT indicado', () => { 
        render( <GifItem title={ title } url={ url } />);
        
        //expect ( screen.getByRole('img').src).toBe( url );
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );

      });

      test('debe de mostrar el titulo en el componente', () => { 

        render(<GifItem title={ title } url={ url } />);
        expect( screen.getByText( title )).toBeTruthy();
       });

 });