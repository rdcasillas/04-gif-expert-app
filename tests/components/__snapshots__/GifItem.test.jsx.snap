// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball"
      src="https://gifexpertapp.com/dragon-ball.gif"
    />
    <p>
      Dragon Ball
    </p>
  </div>
</div>
`;
